### PR TITLE
Adding query plan diagnostics context in feed response

### PR DESCRIPTION
**Pull request by simplynaveen20**
Currently there is no direct way to measure query plan api execution timing for query apis.
This pr will add the query plan diagnostics on first feed response, which will be useful in debugging issue for query apis.
Below is the sample feedResponse diagnostics with newly added query plan context in bold

userAgent=Windows10/10.0 JRE/11.0.6 cosmosdb-java-sdk/4.3.0-beta.1
**QueryPlan Start Time (UTC)=18:41:22:3488
QueryPlan End Time (UTC)=18:41:24:7870
QueryPlan Duration (ms)=2438**
0=Retrieved Document Count                 :              50              
Retrieved Document Size                  :           15150 bytes        
Output Document Count                    :              50              
Output Document Size                     :           15249 bytes        
Index Utilization                        :          100.00 %            
Total Query Execution Time               :        0.590000 milliseconds 
Query Preparation Times 
    Query Compilation Time               :        0.060000 milliseconds 
    Logical Plan Build Time              :        0.040000 milliseconds 
    Physical Plan Build Time             :        0.060000 milliseconds 
    Query Optimization Time              :        0.000000 milliseconds 
  Index Lookup Time                      :        0.000000 milliseconds 
  Document Load Time                     :        0.160000 milliseconds 
Runtime Execution Times 
    Query Engine Times                   :        0.030000 milliseconds 
    System Function Execution Time       :        0.000000 milliseconds 
    User-defined Function Execution Time :        0.000000 milliseconds 
  Document Write Time                    :        0.030000 milliseconds 
Client Side Metrics 
  Retry Count                            :               0              
  Request Charge                         :            6.34 RUs          
 
Partition Execution Timeline 
┌────────────┬────────────────────────────────────┬────────────────┬────────────────┬─────────────┬───────────────────┬───────────┐ 
│Partition Id│                         Activity Id│Start Time (UTC)│  End Time (UTC)│Duration (ms)│NUMBER of Documents│Retry Count│ 
├────────────┼────────────────────────────────────┼────────────────┼────────────────┼─────────────┼───────────────────┼───────────┤ 
│           0│1bbdc899-d1cb-11ea-a112-93d68726b008│   18:41:26:2240│   18:41:27:1850│        961.0│                 50│          0│ 
└────────────┴────────────────────────────────────┴────────────────┴────────────────┴─────────────┴───────────────────┴───────────┘ 
 
Scheduling Metrics 
┌────────────┬──────────────────┬─────────────┬──────────────┬────────────────────┬─────────────────────┐ 
│Partition Id│Response Time (ms)│Run Time (ms)│Wait Time (ms)│Turnaround Time (ms)│NUMBER of Preemptions│ 
├────────────┼──────────────────┼─────────────┼──────────────┼────────────────────┼─────────────────────┤ 
│           0│              1318│          962│          1323│                2285│                    1│ 
└────────────┴──────────────────┴─────────────┴──────────────┴────────────────────┴─────────────────────┘

### CODE REVIEW

**Comment by simplynaveen20** [comment]
I will run the perf for queries to find any perf regression

**Comment by simplynaveen20** [comment]
> I will run the perf for queries to find any perf regression

No perf regression detected

  | Number of operation | 100000 |   |   |   |  
-- | -- | -- | -- | -- | -- | --
  |   |   |   |   |   |  
  |   | Master |   |   | QueryPlan PR |  
  |   | Throughput | Totaltime(sec) |   | Throughput | Totaltime(sec)
1 |   | 85.47 | 1,170 |   | 86.27 | 1159
2 |   | 83.38 | 1,199 |   | 82.8 | 1207
3 |   | 84.88 | 1178 |   | 84.05 | 1189
4 |   | 84.95 | 1177 |   | 85.61 | 1168
5 |   | 85.23 | 1173 |   | 85.73 | 1166

**Comment by kushagraThapar** [comment]
@simplynaveen20 - please rebase this with latest master, as there is a fix for spring emulator test.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/FeedResponseDiagnostics.java [hunk 1]
```diff
@@ -4,15 +4,21 @@
...
+import java.time.Duration;
 import java.util.Map;
 
 /**
  * The type Feed response diagnostics.
  */
 public final class FeedResponseDiagnostics {
 
+    private final static String EQUALS = "=";
+    private final static String QUERY_PLAN = "QueryPlan ";
```
[the remark this answers is not in the document]
Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/FeedResponseDiagnostics.java]: Added another constant. Done

File: sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptionCosmosAsyncContainer.java [hunk 2]
```diff
@@ -433,7 +433,10 @@ ItemDeserializer getItemDeserializer() {
                         }
 
                     ).collectList().map(itemList ->
-                        ModelBridgeInternal.createFeedResponseWithQueryMetrics(itemList, page.getResponseHeaders(), BridgeInternal.queryMetricsFromFeedResponse(page))
+                        ModelBridgeInternal.createFeedResponseWithQueryMetrics(itemList,
+                            page.getResponseHeaders(),
+                            BridgeInternal.queryMetricsFromFeedResponse(page),
+                            ModelBridgeInternal.getQueryPlanDiagnosticsContext(page))
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptionCosmosAsyncContainer.java:439]: nice. thank you.

**Pull request merged**